### PR TITLE
Integrates sumeragi

### PR DESCRIPTION
**Pull request by eadventurous**
### Description of the Change
Integration of sumeragi into the main iroha loop.
Also improves performance of `sort_peers`

### CODE REVIEW

File: iroha/src/kura.rs [hunk 1]
```diff
@@ -38,20 +38,22 @@ impl Kura {
         Ok(())
     }
 
-    /// Methods consumes new validated block and atomically stores and caches it.
-    pub async fn store(
-        &mut self,
+    pub fn create_block(
```
Comment (humb1t) [inline iroha/src/kura.rs:41]: Such a method should be inside `Block`'s impl block.

File: iroha/src/lib.rs [hunk 2]
```diff
@@ -75,31 +75,34 @@ impl Iroha {
         );
         let (public_key, private_key) =
             crypto::generate_key_pair().expect("Failed to generate key pair.");
+        let kura = Arc::new(Mutex::new(Kura::new(
+            config.mode,
+            Path::new(&config.kura_block_store_path),
+            blocks_sender,
+        )));
         //TODO: get peers from json and blockchain
+        let this_peer = PeerId {
```
Comment (humb1t) [inline iroha/src/lib.rs:84]: let's rename it to something like `iroha_peer_id`?

File: iroha/src/lib.rs [hunk 3]
```diff
@@ -149,19 +149,13 @@ impl Iroha {
...
-                                .await
-                                .sign_transactions(transactions)
-                                .await
-                                .expect("Failed to sign transactions."),
-                            &*world_state_view.lock().await,
-                        )
-                        .await
-                        .expect("Failed to accept transactions into blockchain.");
+                    let sumeragi = sumeragi.lock().await;
+                    if let sumeragi::Role::Leader = sumeragi.role() {
```
Comment (humb1t) [inline iroha/src/lib.rs:153]: make a `use sumeragi::Role`

File: iroha/src/sumeragi.rs [hunk 4]
```diff
@@ -118,33 +116,36 @@ impl Sumeragi {
...
 
-    pub async fn on_block_created(&self, block: Block) -> Result<Block, String> {
+    pub async fn validate_and_store(
+        &self,
+        transactions: Vec<Transaction>,
+        wsv: Arc<Mutex<WorldStateView>>,
+    ) -> Result<(), String> {
+        let transactions = self.sign_transactions(transactions).await?;
+        let block = Kura::create_block(transactions, &*wsv.lock().await);
+        if self.sorted_peers.len() < 2 {
```
Comment (humb1t) [inline iroha/src/sumeragi.rs:139]: maybe it can be a good practice to extract int into some named constant?

File: iroha/src/kura.rs (hunk 1 above)
Comment (eadventurous) [reply iroha/src/kura.rs:41]: Yes, thanks, I was also thinking about it

File: iroha/src/sumeragi.rs (hunk 4 above)
Comment (eadventurous) [reply iroha/src/sumeragi.rs:139]: It is a special case when we only have one peer, so to my mind using constant here will only make it harder to understand. Maybe I should just replace it with `.len()==1`. As right now it's impossible to create Sumeragi with 0 peers, it has to have itself at least.

**Pull request merged**